backend/routes/users.py

```diff
@@ -38,12 +54,33 @@ async def get_user_from_headers(headers: dict[str, str], db: AsyncSession):
             raise HTTPException(
                 status_code=status.HTTP_401_UNAUTHORIZED,
             )
+
+    # Try to find existing user
     result = await db.execute(
         select(models.User).where(
             (models.User.username == username) | (models.User.email == email)
         )
     )
-    return result.scalar_one_or_none()
+    user = result.scalar_one_or_none()
+
+    # If user doesn't exist, create them (trusting OAuth proxy validation)
+    if not user:
+        log.info(
+            f"Adding user from OAuth proxy headers: username={username}, email={email}"
+        )
+        user = models.User(
+            username=username
+            or email,  # Use email as username if username not provided
+            email=email or username,  # Use username as email if email not provided
```

Comment:
I don't think this will work - if email is not a valid email, it will break

Reply:
I am not sure I understand.

Here the assumption is that we are getting username and email from headers set by oauth proxy which come from the google IDP. They should be valid for our use case. The caveat being that /profile api should not be accessible without going through oauth proxy as that would allow setting headers arbitrarily.

Reply:
Thanks for pointing it, I have fixed the logic to explicitly fail of user or email headers are missing. It should look better now 👍